### DISCUSSION TITLE
tdslite sketch for ESP8266 NodeMcu V3 - Errors

### DISCUSSION THREAD

**Discussion by Jalberto1964** [tdslite/tdslite#19] [Q&A] [answered]
Arduino IDE 2.1.0
Mode MCU 1.0 (ESP-12E Module)
tdslite 0.2.0

Depending on the ESP8266 Community version (2.7.4 or 3.1.2) we have a compilation error for the 2.7.4 version and a runtime error for the 3.1.2 version (as per attachments).

https://github.com/mustafakemalgilor/tdslite/assets/137678013/7beb9d31-c7d5-49a3-8e0c-29db49f57ef1

[ESP8266 Community 2_7_4 Error.txt](https://github.com/mustafakemalgilor/tdslite/files/11859021/ESP8266.Community.2_7_4.Error.txt)

**Comment by xmkg** [comment, marked as answer]
Hello @Jalberto1964,

The first problem is a compiler problem, so compiler is failing to parse the code properly and crashing.

As for the second one, I see no runtime errors in the recorded video. All the printed messages indicate a normal operation. The `row field N` messages also confirm that the library is able to retrieve data from the connected database. tdslite may print additional debug messages like `netbuf: consumed n , inuse n free n`, or `not enough bytes` when `TDSL_DEBUG_PRINT_ENABLED` macro is defined. tdslite may have to deal with fragmented data more in a wireless connection, and the frequency of these kind of messages depend on the connection quality.

You can simply undefine the `TDSL_DEBUG_PRINT_ENABLED` to get rid of those extra messages.

**Comment by Anuradhamk** [comment]
Hello I'm using the example for NodeMCU V3 (esp8266) board mentioned on this link : https://github.com/tdslite/tdslite/blob/main/tests/sketches/esp/esp.cpp.
Software version: Arduino IDE 2.2.1, TDSL library: tdslite: 0.4.0

I'm getting following error:

"In file included from c:\Users\kamblea1\Documents\Arduino\libraries\tdslite\src/tdslite/detail/tdsl_tds_context.hpp:36:0, 

                 from c:\Users\kamblea1\Documents\Arduino\libraries\tdslite\src/tdslite/detail/tdsl_driver.hpp:17, 

                 from c:\Users\kamblea1\Documents\Arduino\libraries\tdslite\src/tdslite/tdslite.hpp:18, 

                 from c:\Users\kamblea1\Documents\Arduino\libraries\tdslite\src/tdslite.h:22, 

                 from C:\Users\kamblea1\Documents\Arduino\TDSL_example_MSSQL_esp8266\TDSL_example_MSSQL_esp8266.ino:22: 

c:\Users\kamblea1\Documents\Arduino\libraries\tdslite\src/tdslite/detail/tdsl_net_rx_mixin.hpp:31:35: internal compiler error: in cp_parser_abort_tentative_parse, at cp/parser.c:23793 

             static_cast<void *>(0))); 

                                   ^ 

libbacktrace could not find executable to open 

Please submit a full bug report, 

with preprocessed source if appropriate. 

See <http://gcc.gnu.org/bugs.html> for instructions. 

  

exit status 1 

  

Compilation error: exit status 1"

I have commented out these lines:
"#define TDSL_DEBUG_PRINT(FMTSTR, ...)       SERIAL_PRINTF(FMTSTR, ##__VA_ARGS__)
#define TDSL_DEBUG_PRINTLN(FMTSTR, ...)     SERIAL_PRINTLNF(FMTSTR, ##__VA_ARGS__)
#define TDSL_DEBUG_PRINT_U16_AS_MB(U16SPAN) SERIAL_PRINT_U16_AS_MB(U16SPAN)"
Could you please guide me on how to resolve the compilation error?
Thank you so much in advance.

**Reply by xmkg** [reply]
> c:\\Users\\kamblea1\\Documents\\Arduino\\libraries\\tdslite\\src/tdslite/detail/tdsl_net_rx_mixin.hpp:31:35: internal compiler error: in cp_parser_abort_tentative_parse, at cp/parser.c:23793

Hello,

This seem to be a compiler bug rather than a library one. Please try with a different toolchain version.
